### DISCUSSION TITLE
[Question] How is prompt attention different compare to A1111's?

### DISCUSSION THREAD

**Discussion by ChiNoel-osu** [vladmandic/sdnext#3041] [Q&A] [answered]
Recently I switched to SD.Next from A1111's webui because I want to try new stuffs and I heard SD.Next is faster and yes it's sooooo much faster.

But I couldn't help but notice that prompting is a bit different in SD.Next, specifically applying more attention to the prompt e.g. `a (cat:1.5) running` can drastically affect the output of the model (often negatively, looked a lot like when too much attention was applied). But this wasn't the case back in A1111's.
If I remove all the extra attentions from the prompt or just apply a little bit e.g. `a (cat:1.1) running`, the result looks good.
So I guess in SD.Next the attention is treated more "seriously", but why?

People are saying the Diffusers backend is different but can anyone explain how it's different and **what're the best practices for applying attention** in SD.Next?

_I mainly use SDXL models._

**Comment by vladmandic** [comment, marked as answer]
rule-of-a-thumb is that attention sum should be neutral. if you're bringing attention to something and not reducing attention from something else your prompt becomes unbalanced and can result in what appears as burnt image.

`a (cat:1.5) and a (dog:0.5)` would be an example of balanced prompt.

you can get away with slight unbalancing, so `a (cat:1.1)` is still fine, but `a (cat:1.5)` means majority of prompt is under very strong attention and that doesnt look nicely.

even worse if you apply it to negative prompt. 

contrary common belief, negative prompts don't work by preventing something to start with - they ADD it first and then steer away from it.
so if you add `a (cat:1.5)`, you just added a ghost of a cat to image in the first step and then veeeery strongly going in a direction opposite of a cat - and that limits models behavior significantly. so my advice is don't use negative prompts for things that don't appear in general.

but in general, with modern prompt parser and modern models, there is faaar less need for attention than it used to be.

**Reply by drax-xard** [reply]
Oh wow, that clears up so much that I had misunderstood about prompt attn. Thanks!
